dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+    return output_layout_;
+  }
+
+  const std::vector<int64_t> &GetInChunkOffsets() const {
+    return offsets_;
+  }
+
+  const std::vector<size_t> &GetInChunkSizes() const {
+    return sizes_;
+  }
+
+  bool HasChunks() const {
+    return HasExplicitChunkOffsets() || HasExplicitChunkSizes();
+  }
+
+ private:
+  bool HasExplicitChunkOffsets() const {
+    return chunk_offsets_.HasExplicitValue();
+  }
+
+  bool HasExplicitChunkSizes() const {
+    return chunk_sizes_.HasExplicitValue();
+  }
+
+  void SetupOutputLayout() {
+    if (layout_.empty()) {
+      return;
+    }
+    auto sample_dim = GetOutputShape().sample_dim() - HasChunks();
+    DALI_ENFORCE(
+        layout_.size() == sample_dim,
+        make_string("The size of inflated chunks' layout must match the number of extents in "
+                    "the output shape. However, got the layout `",
+                    layout_, "` while the chunks output shape has ", sample_dim, " extent(s)."));
+    if (!HasChunks()) {
+      output_layout_ = layout_;
+    } else {
+      assert(sequence_extent_.size() == 1);
+      output_layout_ = sequence_extent_ + layout_;
+    }
+  }
+
+  void SetupOffsetsAndSizes(const Workspace &ws) {
+    const auto &in_shape = ws.GetInputShape(0);
+    int batch_size = in_shape.num_samples();
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      DALI_ENFORCE(in_shape[sample_idx].num_elements() > 0,
+                   make_string("Input sample cannot be empty, but the sample at index ", sample_idx,
+                               " has volume 0."));
+    }
+    if (HasChunks()) {
+      AcquireInferOffsetsSizes(ws);
+    } else {
+      chunks_per_sample_ = uniform_list_shape(batch_size, TensorShape<1>{1});
+      offsets_.resize(batch_size, 0);
+      sizes_.clear();
+      sizes_.reserve(batch_size);
+      for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+        sizes_.push_back(in_shape[sample_idx].num_elements());
+      }
+    }
+  }
+
+  void AcquireInferOffsetsSizes(const Workspace &ws) {
+    assert(HasChunks());
+
+    const auto copy_to_vector = [](auto &v, const auto &tlv) {
+      v.clear();
+      v.reserve(tlv.shape.num_elements());
+      for (int sample_idx = 0; sample_idx < tlv.shape.num_samples(); sample_idx++) {
+        auto num_elements = tlv.shape[sample_idx].num_elements();
+        const auto *chunks_data = tlv.data[sample_idx];
+        v.insert(v.end(), chunks_data, chunks_data + num_elements);
+      }
+    };
+
+    const auto validate_offset = [](auto sample_idx, auto sample_num_bytes, auto chunk_offset) {
+      DALI_ENFORCE(chunk_offset >= 0,
+                   make_string("Input chunks offsets must be non-negative. Got ", chunk_offset,
+                               " offset for sample of idx ", sample_idx, "."));
+      DALI_ENFORCE(
+          chunk_offset < sample_num_bytes,
+          make_string("Input chunks offsets cannot exceed the sample size. Got chunk offset ",
```

Comment:
Strictly speaking, the offset == sample_num_bytes doesn't _exceed_ the sample size.
```suggestion
          make_string("Input chunks offsets cannot point past the last byte. Got chunk offset ",
```

Reply:
done